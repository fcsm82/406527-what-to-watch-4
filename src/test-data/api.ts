import axios from 'axios';

export const api = axios.create({
  baseURL: `https://4.react.pages.academy/wtw`,
  timeout: 5000,
  withCredentials: true,
});
