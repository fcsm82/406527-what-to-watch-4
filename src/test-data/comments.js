export const comment =
{
  id: 4563456345,
  author: {
    id: 1223,
    name: `Amanda Greever`,
  },
  rating: 8.0,
  text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  date: `2019-05-08T14:13:56.569Z`,
};

export const comments =
[{
  id: 4563456345,
  author: {
    id: 1223,
    name: `Amanda Greever`,
  },
  rating: 8.0,
  text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  date: `2019-05-08T14:13:56.569Z`,
},
{
  id: 123124,
  author: {
    id: 12415252,
    name: `Amanda Greever`,
  },
  rating: 7.0,
  text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  date: `2019-05-08T14:13:56.569Z`,
},
{
  id: 23462324631461,
  author: {
    id: 1234,
    name: `Amanda Greever`,
  },
  rating: 8.0,
  text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  date: `2019-05-08T14:13:56.569Z`,
}];
