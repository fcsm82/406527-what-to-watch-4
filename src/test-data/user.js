import {BASE_URL} from "../consts.js";

export const user = {
  id: 1213,
  email: `john.doe@johndoe.com`,
  name: `John Doe`,
  avatarSrc: `${BASE_URL}$/avatarSrc`,
};
