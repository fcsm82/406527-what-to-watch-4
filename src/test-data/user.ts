import {User} from "../types";

export const user: User = {
  id: 1213,
  email: `john.doe@johndoe.com`,
  name: `John Doe`,
  avatarSrc: `https://4.react.pages.academy/wtw/asda.jpg`
};
