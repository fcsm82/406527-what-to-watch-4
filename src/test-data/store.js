import NameSpace from "../store/reducers/name-space.js";
import {promoFilm, films} from "../test-data/films.js";
import {user} from "../test-data/user.js";

export const testStore = {
  [NameSpace.DATA]: {
    isAppLoading: true,
    isAppError: false,
    films,
    promoFilm,
    favoriteFilms: [],
    comments: []
  },
  [NameSpace.FILMS]: {
    shownFilms: 8,
    activeFilter: `All genres`,
  },
  [NameSpace.USER]: {
    error: null,
    authorizationStatus: `No auth`,
    authorizationError: false,
    user
  },
};

