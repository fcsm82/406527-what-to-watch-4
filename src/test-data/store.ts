import NameSpace from "../store/reducers/name-space";
import {promoFilm, films} from "../test-data/films";
import {user} from "../test-data/user";

export const testStore = {
  [NameSpace.DATA]: {
    isAppLoading: true,
    isAppError: false,
    films,
    promoFilm,
    favoriteFilms: [],
    comments: []
  },
  [NameSpace.FILMS]: {
    shownFilms: 8,
    activeFilter: `All genres`,
  },
  [NameSpace.USER]: {
    error: null,
    authorizationStatus: `No auth`,
    authorizationError: false,
    user
  },
};

