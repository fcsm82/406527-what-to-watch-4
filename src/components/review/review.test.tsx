import * as React from "react";
import * as renderer from "react-test-renderer";

import {Provider} from "react-redux";

import configureStore from "redux-mock-store";

import Review from "./review";

import {comment} from "../../test-data/comments";
import {testStore} from "../../test-data/store";

const mockStore = configureStore([]);

const store = mockStore(testStore);

it(`Should review render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <Review comment = {comment}/>
        </Provider>
        , {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

  expect(tree).toMatchSnapshot();
});
