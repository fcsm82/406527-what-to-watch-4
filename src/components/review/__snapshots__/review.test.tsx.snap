// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should review render correctly 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
      style={
        Object {
          "overflow": "hidden",
          "textOverflow": "ellipsis",
        }
      }
    >
      I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        Amanda Greever
      </cite>
      <time
        className="review__date"
        dateTime="2019-05-08"
      >
        May 08, 2019
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    8
  </div>
</div>
`;
